### PR TITLE
contextual logging: beta in Kubernetes 1.30

### PR DESCRIPTION
**Pull request by pohly**
Related-to: https://github.com/kubernetes/enhancements/issues/3077

/hold

https://github.com/kubernetes/kubernetes/pull/122589 needs to be merged first.

### CODE REVIEW

File: content/en/docs/concepts/cluster-administration/system-logs.md
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/system-logs.md:136]:
```suggestion
For Kubernetes {{< skew currentVersion >}}, this is gated behind the `ContextualLogging`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) and is
```

File: content/en/docs/concepts/cluster-administration/system-logs.md
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/system-logs.md:1]: At some place in this page you could add a note that `ContextualLogging` was previously named `StructuredLogging` (assuming that is true).

**Review by sftim — COMMENTED** [review]
Please also update `content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md`

**Comment by drewhagen** [comment]
Hello @pohly :wave: please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday March 12th 2024 18:00 PST. Thank you!

File: content/en/docs/concepts/cluster-administration/system-logs.md
Comment (pohly) [reply content/en/docs/concepts/cluster-administration/system-logs.md:1]: That was actually a mistake in the previous doc. The feature gate has always been `ContextualLogging`.

File: content/en/docs/concepts/cluster-administration/system-logs.md
Comment (pohly) [reply content/en/docs/concepts/cluster-administration/system-logs.md:136]: Done.

I also updated the feature gate description.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md
Comment (pohly) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md:18]: Note that I switched from "if you enable" to "if you disable" here because that is what users would do now.

Also note that when converting existing log calls we try to avoid information loss should the feature get disabled. But in new code that I recently wrote for DRA, I am assuming that the feature remains enabled and avoid logging information both via context and as parameters.

I expect this to happen more often going forward because the duplication is annoying.

**Comment by drewhagen** [comment]
Hello @pohly 👋, it looks like you've already engaged feedback in review of this PR. Great job! Keep this up. 🚀

@alculquicondor @liggitt @serathius @ehashman @thockin
When one of y'all have a moment, please give a review for technical accuracy. A friendly reminder that this PR needs doc review complete by Doc Freeze on **_March 26th 18:00 PT_** to get this into the release.

It also looks like there is a hold here and a PR that has been merged. Can this hold be removed?

Thanks!

**Comment by pohly** [comment]
There's not much technical content that needs to be reviewed here. I think it's fine if the docs team does a LGTM + approve for the wording and formatting.

/hold cancel

Implementation was merged, PR can be merged, too.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md
Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md:18]: We always write descriptions as what happens when the feature is _enabled_.
Try
```markdown
Enables additional detail in log output from Kubernetes components that
support writing structure logs.
```

**Review by sftim — CHANGES_REQUESTED** [review]
One thing to fix, otherwise LGTM

File: content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md
Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md:18]: Think about the graduation to stable; Kubernetes still has the feature, but now you cannot disable it. We want a feature description that makes sense for beta and for stable.

Comment (pohly) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/contextual-logging.md:18]: Makes sense. I used the text that I had in the main file because this works regardless of the log output format.

> Enables extra details in log output of Kubernetes components that support
> contextual logging.

File: content/en/docs/concepts/cluster-administration/system-logs.md
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/system-logs.md:123]: (nit / good to fix before GA / yes, I know the rest of the file is also not following the style guide here)
```suggestion
### Contextual logging
```

**Review by sftim — COMMENTED** [review]
/lgtm

for docs

/approve
I don't think this needs a technical review before it merges.

**Pull request merged**
